### DISCUSSION TITLE
KeypadLinc 9 buttons in HA?

### DISCUSSION THREAD

**Discussion by johnsturgeon** [TD22057/insteon-mqtt#490] [Q&A] [answered]
I'm curious why a 6 button KeypadLinc shows 9 buttons in Home Assistant.
<img width="667" alt="image" src="https://user-images.githubusercontent.com/9746310/156815757-0a5c3713-eac4-414d-b2ae-9edc0400ad82.png">

I need to define a link between my KeypadLinc button 'A' (button 2?) and another light, but I don't really know which button is which in Home Assistant

UPDATE (not an answer really..): After playing around, it seems that buttons 3,4,5,6 map to A,B,C,D and buttons 2,7,8,9 seem to be complete mysteries to me still

**Comment by kpfleming** [comment, marked as answer]
This is the current expected behavior (although I'm working on some improvements in this area).

This code supports both 6- and 6-button KeypadLinc devices, and those devices can be configured in 'detached load' mode (where the built-in dimmer or relay is *not* controlled by the built-in buttons, but controlled independently). As a result the code provides 9 entities, which covers the maximum number that might be available.

As you've noted, on a 6-button unit buttons 3-6 map to A-D; button 1 is for the built-in dimmer (or relay) "ON" button, as is button 2 if I remember correctly. I don't know how the remaining button ("OFF") is mapped, nor how they are mapped on an 8-button unit as I don't have any of those.

If you'd like to change the HA discovery mappings to make them map more closely to your 6-button device, you can use the information at this [link](https://github.com/TD22057/insteon-mqtt/blob/master/docs/discovery.md#six-button-keypadlinc).

**Reply by kpfleming** [reply]
Also, the button mappings are documented [here](https://github.com/TD22057/insteon-mqtt/blob/c8e0ce36a028b1296d7b466a50ba75471f0dbcdc/insteon_mqtt/data/config-base.yaml#L1318).
